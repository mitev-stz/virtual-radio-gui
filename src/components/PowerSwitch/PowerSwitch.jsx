import React from 'react';
import "./assets/styles/powerSwitch.css";

class PowerSwitch extends React.Component {
  render () {
    const{handleToggleSwitchAction} = this.props;
      return (
          <span className="power-switch" placeholder="power-switch">
              <label className="switch">
                <input data-testid= "checkbox"  type="checkbox" onClick = {handleToggleSwitchAction}></input>
                <span className="slider round"></span>
              </label>
          </span>
        );
  }
  getBtnOnClasses(){
    let classes = "btn powerButton-on ";
    classes+= this.props.isRadioLive? " btn powerButton-on-active" : "";
    return classes;
  }
}

export default PowerSwitch;
