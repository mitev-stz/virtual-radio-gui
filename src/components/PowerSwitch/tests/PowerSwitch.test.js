import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PowerSwitch from '../PowerSwitch';
import TestRenderer  from 'react-test-renderer';

it("renders correctly", () =>{
  const {getByPlaceholderText} = render(<PowerSwitch/>)
  const linkElement1 = getByPlaceholderText("power-switch");
  expect(linkElement1).toBeInTheDocument();
});

it("updates internal state on buttons clicked", () =>{
  const{queryByTestId} = render(<PowerSwitch/>);
  const component = TestRenderer.create(<PowerSwitch></PowerSwitch>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  const buttonOn = queryByTestId("checkbox");
  fireEvent.click(buttonOn);
  expect(tree).toMatchSnapshot();
  fireEvent.click(buttonOn);
  expect(tree).toMatchSnapshot();
});
