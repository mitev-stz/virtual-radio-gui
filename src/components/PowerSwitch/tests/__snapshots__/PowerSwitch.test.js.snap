// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updates internal state on buttons clicked 1`] = `
<span
  className="power-switch"
  placeholder="power-switch"
>
  <label
    className="switch"
  >
    <input
      data-testid="checkbox"
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
</span>
`;

exports[`updates internal state on buttons clicked 2`] = `
<span
  className="power-switch"
  placeholder="power-switch"
>
  <label
    className="switch"
  >
    <input
      data-testid="checkbox"
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
</span>
`;

exports[`updates internal state on buttons clicked 3`] = `
<span
  className="power-switch"
  placeholder="power-switch"
>
  <label
    className="switch"
  >
    <input
      data-testid="checkbox"
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
</span>
`;
